listener/listener.go

```diff
@@ -175,16 +216,62 @@ func (l *listener) createVulnerabilityOccurrences(event *harbor.Event) ([]*grafe
 	return occurrences, nil
 }
 
-func resourceUri(event *harbor.Event) string {
-	base := strings.Split(event.Data.Resources[0].ResourceUrl, ":")[0]
+func (l *listener) createNoteForReport(ctx context.Context, event *harbor.Event, resource *harbor.Resource, report *harbor.Report) (string, error) {
+	artifactUrl, err := l.harborClient.GetArtifactUrl(event.Data.Repository.Namespace, event.Data.Repository.Name, resource.Digest)
+	if err != nil {
+		return "", err
+	}
 
-	return fmt.Sprintf("%s@%s", base, event.Data.Resources[0].Digest)
+	note, err := l.rodeClient.CreateNote(ctx, &pb.CreateNoteRequest{
+		Note: &grafeas_go_proto.Note{
+			ShortDescription: "Harbor Vulnerability Scan",
+			LongDescription:  fmt.Sprintf("Harbor Vulnerability Scan by %s/%s (%s)", report.Scanner.Vendor, report.Scanner.Name, report.Scanner.Version),
+			Kind:             common_go_proto.NoteKind_DISCOVERY,
+			RelatedUrl: []*common_go_proto.RelatedUrl{
+				{
+					Label: "Artifact URL",
+					Url:   artifactUrl,
+				},
+			},
+			Type: &grafeas_go_proto.Note_Discovery{
+				Discovery: &discovery_go_proto.Discovery{
+					AnalysisKind: common_go_proto.NoteKind_VULNERABILITY,
+				},
+			},
+		},
+		NoteId: fmt.Sprintf("harbor-scan-%s", report.ReportId),
```

Comment:
Another demo-related followup is that the sample policy we have for Harbor at the moment looks at the [exact note name](https://github.com/rode/demo/blob/cc937553b24c950c1f1d7a0a4a84c3e957e9872b/tf/modules/rode/policies/harbor.rego#L12), that'll have to change to `startswith` instead